src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1373,13 +1373,13 @@ private TypeDeclarationSyntax ParseClassOrStructOrInterfaceDeclaration(SyntaxLis
             Debug.Assert(!IsInAsync);
 
             var classOrStructOrInterface = this.EatToken();
+
             var saveTerm = _termState;
             _termState |= TerminatorState.IsPossibleAggregateClauseStartOrStop;
             var name = this.ParseIdentifierToken();
             var typeParameters = this.ParseTypeParameterList();
-
-            _termState = saveTerm;
             var baseList = this.ParseBaseList();
+            _termState = saveTerm;
```

Comment:
this is the actual change.  as long we we're in the base list, we stay in the `PossibleAggregateClauseStartOrStop` state so that we consider tokens like `{` something that should stop us from parsing the base-list.